src/components/transactions/GroupedTxListItems/WillReplaceProvider.tsx

```diff
@@ -0,0 +1,53 @@
+import { createContext, useMemo, useState, type Dispatch, type ReactElement, type SetStateAction } from 'react'
+import type { Transaction } from '@gnosis.pm/safe-react-gateway-sdk'
+
+import { useAppSelector } from '@/store'
+import { selectPendingTxs } from '@/store/pendingTxsSlice'
+
+export const WillReplaceContext = createContext<{
+  willReplace: string[]
+  setWillExecute: Dispatch<SetStateAction<string | undefined>>
+}>({
+  willReplace: [],
+  setWillExecute: () => {},
+})
+
+// Used for striking through transactions that will be replaced
+export const WillReplaceProvider = ({
```

Comment:
The `willReplace` terminology isn't very clear out of context (pun not intended).

I would call the provider `ReplaceTxHoverProvider` or some such (like the BatchHoverProvider).

Instead of `willReplace` in the return value: 'replacedTxIds`.